docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -529,3 +529,166 @@ individual step parameter must be set when using this method, or else the coded
 defaults will be used, which may be inappropriate for the dataset being processed.
 
 See :ref:`call_examples` for more information.
+
+
+.. _multiprocessing:
+
+Multiprocessing
+===============
+
+Multiprocessing is supported to speed up certain computationally-intensive steps
+in the pipeline, including the :ref:`jump detection <jump_step>`,
+:ref:`ramp fitting <ramp_fitting_step>`, and
+:ref:`WFSS contamination correction <wfss_contam_step>` steps. The examples below show how
+multiprocessing can be enabled for these steps, as well as how to set up
+multiprocessing to simultaneously run the entire pipeline on multiple observations.
+
+Python's multiprocessing module explicitly imports and executes a script's
+`__main__` with each and every worker. If `__main__` is not present the behavior is
+undefined. Hence, Python will crash unless the multiprocess code in enclosed in a
+`__main__` block like this:
+
+
+::
+
+    import sys
+
+    def main():
+        [code used in multiprocessing]
+
+    if __name__ = '__main__':
+        sys.exit(main())
+
```

Comment:
```suggestion
Since the pipeline uses multiprocessing it is critical that any code using the pipeline adhere to the guidelines described in the `python multiprocessing documentation <https://docs.python.org/3/library/multiprocessing.html#multiprocessing-programming>`_. The pipeline uses the `forkserver` start method internally and it is recommended that any multiprocessing scripts that use the pipline use the same start. As detailed in the `python documentation <https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods>`_ this will require that code be "protected" with a ``if __name__ == '__main__':`` check as follows

::

    if __name__ = '__main__':
        [code used in multiprocessing]

```

What do you think about changing this to link to the python docs section that provides guidance for multiprocessing? I think the links could use the intersphinx mapping (but I have little confidence that I would get the mapping right on the first try). For example, for the suggestion above the following might be replaceable:
```
`python multiprocessing documentation <https://docs.python.org/3/library/multiprocessing.html#multiprocessing-programming>`_
```
with:
```
:ref:`python multiprocessing documentation <python:multiprocessing-programming>`
```